llama-stack-client-kotlin-client-local/src/main/kotlin/com/llama/llamastack/client/local/util/ResponseUtil.kt

```diff
@@ -0,0 +1,61 @@
+package com.llama.llamastack.client.local.util
+
+import com.llama.llamastack.core.JsonValue
+import com.llama.llamastack.models.CompletionMessage
+import com.llama.llamastack.models.InferenceChatCompletionResponse
+import com.llama.llamastack.models.ToolCall
+import java.util.UUID
+
+fun buildInferenceChatCompletionResponse(
+    response: String,
+    stats: Float
+): InferenceChatCompletionResponse {
+    // check for prefix [ and suffix ] if so then tool call.
+    // parse for "toolName", "additionalProperties"
+
+    var completionMessage =
+        if (response.startsWith("[") && response.endsWith("]")) {
```

Comment:
QQ: @cmodi-meta Is this how LS server end parsing the custom tool call as well?

Reply:
it's similar to how iOS on Llama Stack server does the parsing (https://github.com/meta-llama/llama-stack/blob/main/llama_stack/providers/inline/ios/inference/LocalInferenceImpl/Parsing.swift#L152). This is what Dalton pointed to.